app/store/services/helpers/gps.js

```diff
@@ -48,4 +47,23 @@ const getNearestPoint = (points = [], currentPosition = {}) => {
   return currentPosition;
 };
 
-export default { getDistance, isInArea, getEventDistance, getNearestPoint };
+const computeElevationGain = (points = []) =>
```

Comment:
you should consider a function to get edges from a list of path so you could iterate WITHOUT artifice (id est -> previousPoint)